recipes/cppcommon/all/patches/0001-update-cmakelists-1-0-1-0.patch

```diff
@@ -0,0 +1,139 @@
+From 85122652b618951f1b8fe91d8804d92a74d971be Mon Sep 17 00:00:00 2001
+From: Alejandro Colomar <alx.manpages@gmail.com>
+Date: Mon, 7 Jun 2021 10:27:46 +0200
+Subject: [PATCH] update cmakelists
+
+---
+ CMakeLists.txt | 42 +++++++++++++++++++-----------------------
+ 1 file changed, 19 insertions(+), 23 deletions(-)
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index aa5447a8..10a0065b 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -17,49 +17,36 @@ if(DOXYGEN_FOUND)
+   endif()
+ endif()
+ 
+-# CMake module path
+-set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+-
+-# Compiler features
+-include(SetCompilerFeatures)
+-include(SetCompilerWarnings)
+-include(SetPlatformFeatures)
+-include(SystemInformation)
+-
+ # External packages
+ find_package(Threads REQUIRED)
+ if(UNIX AND NOT APPLE AND NOT MSYS)
+   find_package(LibBFD)
+   find_package(LibDL)
+   find_package(LibRT)
+-  find_package(LibUUID)
++  find_package(libuuid)
+ endif()
+ if(WIN32 OR MSYS)
+   find_package(DbgHelp)
+   find_package(RPC)
+-  find_package(Userenv)
+ endif()
+ 
+-# Modules
+-add_subdirectory("modules")
+-
+ # Link libraries
+ list(APPEND LINKLIBS Threads::Threads)
+ if(UNIX AND NOT APPLE AND NOT MSYS)
+   list(APPEND LINKLIBS ${LIBBFD_LIBRARIES})
+   list(APPEND LINKLIBS ${LIBDL_LIBRARIES})
+   list(APPEND LINKLIBS ${LIBRT_LIBRARIES})
+-  list(APPEND LINKLIBS ${LIBUUID_LIBRARIES})
++  list(APPEND LINKLIBS CONAN_PKG::libuuid)
+ endif()
+ if(WIN32 OR MSYS)
+   list(APPEND LINKLIBS ${DBGHELP_LIBRARIES})
+   list(APPEND LINKLIBS ${RPC_LIBRARIES})
+-  list(APPEND LINKLIBS ${USERENV_LIBRARIES})
++  list(APPEND LINKLIBS userenv)
++  list(APPEND LINKLIBS rpcrt4)
```

Comment:
Is the find `rpc` not getting it?

Reply:
That is a copy&paste of the patch for cppcommon/1.0.0.0.  See <https://github.com/conan-io/conan-center-index/blob/08f8662c2705aed9235ade59cd22b01336ef7f1d/recipes/cppcommon/all/patches/0001-update-cmakelists-1-0-0-0.patch>.

I don't know why it was patched.  I still don't know too much of CMake...

BTW, for the future, having useful commit messages for those patches is a good idea :-)

Reply:
> BTW, for the future, having useful commit messages for those patches is a good idea

Do you know any tools which can check for that? I wish this was something we could enforce

Reply:
I think you could use git hooks to check if any new files with the extension .patch are added.

If so, you could use checkpatch (I haven't used it for this, but for static analysis of .c source files, but it also checks some things about git patches; I'll check) to check that the patches are git patches, and not just diffs.

I never did that, so let me check.  I'll come back with something.

Reply:
Humm, checkpatch complains about patches that don't have a valid 'Signed-off-by: Name <email>' line.

But it doesn't complain about any other metadata of patches.
And the worst part is that that check is only triggered if the patch is a git patch;
if it's just a diff, it doesn't complain, and will only check the source of the patch.

So maybe a simple bash/python script that checks the contents of the patch would serve this purpose.
It should find a line that begins with `/^Subject: /` before it finds a line that begins with `/^diff/`.

The following should be enough:
```bash
alx@lvl01026:~/src/conan/conan-center-index/recipes/cppcommon/all/patches$ cat check_patch_format.sh 
#!/bin/bash
set -Eeuo pipefail;

EX_OK=0;
EX_DATAERR=65;

patch="$1";

< "$patch" \
sed '/^diff/,$d' \
| grep -q '^Subject: ' \
&& exit ${EX_OK};

1>&2 echo "$patch is not a git patch.";
1>&2 echo "Please use git-format-patch(1) to create the patch.";
exit ${EX_DATAERR};

alx@lvl01026:~/src/conan/conan-center-index/recipes/cppcommon/all/patches$ ./check_patch_format.sh 0001-update-cmakelists-1-0-0-0.patch 
0001-update-cmakelists-1-0-0-0.patch is not a git patch.
Please use git-format-patch(1) to create the patch.
alx@lvl01026:~/src/conan/conan-center-index/recipes/cppcommon/all/patches$ echo $?
65
alx@lvl01026:~/src/conan/conan-center-index/recipes/cppcommon/all/patches$ ./check_patch_format.sh 0001-update-cmakelists-1-0-1-0.patch 
alx@lvl01026:~/src/conan/conan-center-index/recipes/cppcommon/all/patches$ echo $?
0

```

Reply:
In the meantime, I'll report that behavior to the maintainers of checkpatch, to see if they want to fix it.

Reply:
Thanks for the idea!

Reply:
We can enforce some extra fields in the `conandata.yml` for the patches (using a hook). We are upgrading Conan in the CI and now it won't fail if there are some extra fields. I'm talking about this kind of information: https://github.com/jgsogo/conan-center-index/pull/4/files